src/controllers/peer-protocol.ts

```diff
@@ -0,0 +1,35 @@
+'use strict'
+
+import * as IlpPacket from 'ilp-packet'
+import IldcpHostController from './ildcp-host'
+import InvalidPacketError from '../errors/invalid-packet-error'
+import reduct = require('reduct')
+
+const PEER_PROTOCOL_CONDITION = Buffer.from('Zmh6rfhivXdsj8GLjp+OIAiXFIVu4jOzkCpZHQ1fKSU=', 'base64')
+const PEER_PROTOCOL_FULFILLMENT = Buffer.alloc(32)
+
+export default class PeerProtocolController {
+  protected ildcpHostController: IldcpHostController
+
+  constructor (deps: reduct.Injector) {
+    this.ildcpHostController = deps(IldcpHostController)
+  }
+
+  async handle (sourceAccount: string, parsedPacket: IlpPacket.IlpPrepare) {
+    if (!PEER_PROTOCOL_CONDITION.equals(parsedPacket.executionCondition)) {
+      throw new InvalidPacketError('condition must be null.')
+    }
+
+    switch (parsedPacket.destination) {
+      case 'peer.config':
```

Comment:
Nitpick: put `'peer.config'` as a constant up at the top of the file

Reply:
I think we should keep it as a literal. The reason we usually try to name constants is because this:

``` js
car.setType(3)
```

is a lot harder to understand than:

``` js
car.setType(TYPE_MODEL_S)
```

But when the constant is a meaningful string like `peer.config`, it's only used in one place in the file and it's clear from the context that it is the distinguisher for detecting ILDCP, then a constant just adds an unnecessary level of indirection imo.